src/Evolution/Systems/ScalarWave/BoundaryCorrections/Factory.hpp

```diff
@@ -0,0 +1,7 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Evolution/Systems/ScalarWave/BoundaryCorrections/BoundaryCorrection.hpp"
```

Comment:
Yes, I don't remember if we every discussed not using this design. I prefer it because I've sometimes changed a header file for 1 domain creater and had something like 150 targets rebuild, all because of the coupling having the includes be in the base class header file.